emulator/blockchain.go

```diff
@@ -585,7 +598,8 @@ func configureFVM(blockchain *Blockchain, conf config, blocks *blocks) (*fvm.Vir
 		fvm.WithAccountStorageLimit(conf.StorageLimitEnabled),
 		fvm.WithTransactionFeesEnabled(conf.TransactionFeesEnabled),
 		fvm.WithReusableCadenceRuntimePool(customRuntimePool),
-		fvm.WithEntropyProvider(&dummyEntropyProvider{}),
+		fvm.WithEntropyProvider(blockchain.entropyProvider),
+		fvm.WithRandomSourceHistoryCallAllowed(true),
```

Comment:
Ideally this would only be enabled for the system transaction.

Reply:
That's a good catch :100: